src/decoder_allocators.hpp

```diff
@@ -0,0 +1,149 @@
+/*
+    Copyright (c) 2007-2015 Contributors as noted in the AUTHORS file
+
+    This file is part of libzmq, the ZeroMQ core engine in C++.
+
+    libzmq is free software; you can redistribute it and/or modify it under
+    the terms of the GNU Lesser General Public License (LGPL) as published
+    by the Free Software Foundation; either version 3 of the License, or
+    (at your option) any later version.
+
+    As a special exception, the Contributors give you permission to link
+    this library with independent modules to produce an executable,
+    regardless of the license terms of these independent modules, and to
+    copy and distribute the resulting executable under terms of your choice,
+    provided that you also meet, for each linked independent module, the
+    terms and conditions of the license of that module. An independent
+    module is a module which is not derived from or based on this library.
+    If you modify this library, you must extend this exception to your
+    version of the library.
+
+    libzmq is distributed in the hope that it will be useful, but WITHOUT
+    ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+    FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public
+    License for more details.
+
+    You should have received a copy of the GNU Lesser General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef ZEROMQ_DECODER_ALLOCATORS_HPP
+#define ZEROMQ_DECODER_ALLOCATORS_HPP
+
+#include <stdlib.h>
+
+#include "err.hpp"
+#include "atomic_counter.hpp"
+
+namespace zmq
+{
+    // Static buffer policy.
+    class c_single_allocator
+    {
+    public:
+        c_single_allocator(size_t bufsize_):
```

Comment:
Hi,

I've created a pull-request for the issues. Thanks for pointing them out. Most 
of them I did just because I tried to adapt to the zeroMQ code which is 
already there, e.g. it includes old-style headers in many places and C-casts 
seem to be the standard.  For my own project, I usually avoid all the old-C-
style stuff and just forbid it, e.g. I would not allow malloc/free or C-style 
casts. 

Best wishes,
  Jens

On Sunday 05 July 2015 13:44:09 Pieter Hintjens wrote:
> > +
> > +#ifndef ZEROMQ_DECODER_ALLOCATORS_HPP
> > +#define ZEROMQ_DECODER_ALLOCATORS_HPP
> > +
> > +#include <stdlib.h>
> > +
> > +#include "err.hpp"
> > +#include "atomic_counter.hpp"
> > +
> > +namespace zmq
> > +{
> > +    // Static buffer policy.
> > +    class c_single_allocator
> > +    {
> > +    public:
> 
> > +        c_single_allocator(size_t bufsize_):
> Sure, "code is not as clean as it could be" is a valid problem. We do this
> all the time. Even a space in the wrong place is worth fixing. We have a
> style guide for the code; you may want to check it.
> http://zeromq.org/docs:style (it can be improved as we need.)
> 
> On Sun, Jul 5, 2015 at 10:27 PM, Thomas Köppe <notifications@github.com>
> 
> wrote:
> > In src/decoder_allocators.hpp
> > 
> > <https://github.com/zeromq/libzmq/pull/1466#discussion_r33898561>:
> > > +
> > > +#ifndef ZEROMQ_DECODER_ALLOCATORS_HPP
> > > +#define ZEROMQ_DECODER_ALLOCATORS_HPP
> > > +
> > > +#include <stdlib.h>
> > > +
> > > +#include "err.hpp"
> > > +#include "atomic_counter.hpp"
> > > +
> > > +namespace zmq
> > > +{
> > > +    // Static buffer policy.
> > > +    class c_single_allocator
> > > +    {
> > > +    public:
> > 
> > > +        c_single_allocator(size_t bufsize_):
> > Well, I'm sure what to do, but it the change by itself wouldn't qualify
> > for a "fix" of a problem of any kind. It's all the little things I would
> > probably just have asked the author to tidy up before merging. There are
> > probably other opportunities for cleanup elsewhere, but then again "don't
> > fix what ain't broken"...
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > <https://github.com/zeromq/libzmq/pull/1466/files#r33898561>.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/zeromq/libzmq/pull/1466/files#r33898657